lib/Doctrine/ORM/Mapping/Driver/AnnotationDriver.php

```diff
@@ -29,8 +31,17 @@
 /**
  * The AnnotationDriver reads the mapping metadata from docblock annotations.
  */
-class AnnotationDriver extends AbstractAnnotationDriver
+class AnnotationDriver implements MappingDriver
 {
+    use ColocatedMappingDriver;
+
+    /**
+     * The annotation reader.
+     *
+     * @var Reader
+     */
+    protected $reader;
```

Comment:
Shall we flag the property as `@internal` and make it private on 3.0?

Reply:
Shouldn't we deprecate `AnnotationDriver` and remove it entirely in 3.0?
Also, I'm wondering why `AttributeDriver` has a `getReader()` method, and is not final :thinking: